pkg/manager/impl/execution_manager.go

```diff
@@ -974,6 +975,7 @@ func (m *ExecutionManager) launchExecutionAndPrepareModel(
 		InputsURI:             inputsURI,
 		UserInputsURI:         userInputsURI,
 		SecurityContext:       executionConfig.SecurityContext,
+		LaunchEntity:          request.Spec.LaunchPlan.ResourceType,
```

Comment:
Instead of getting from request spec . Can we not get this from launchplan object.

Also i think we need the same change in launchSingleTaskExecution

Reply:
The launch plan object is a model which doesn't have a core.identifier. Just curious, what would the difference be if it's the same identifier? 

The same change is already done for single task executions in line 724 above.

Reply:
Wanted to avoid usage of request object beyond a certain scope of the function . Ideally starting of service function is where it should be scoped and since its a large function would be good to get this info from derived object if possible.

I see we have this.
```
launchPlan, err := transformers.FromLaunchPlanModel(launchPlanModel)``` 

We should be able to get this from `launchPlan.Id.ResourceType` right ?

I am fine though we this too.

Reply:
thanks for explaining, updated the usage here